README.md

```diff
@@ -1,14 +1,175 @@
 # svelte-concurrency
 
-Handle async task with ease thanks to `svelte-concurrency`
+Handle async tasks with ease thanks to `svelte-concurrency`.
+
+## What is it?
+
+`svelte-concurrency` supplies a simple way to introduce cancellable concurrency into your app. Not only do they provide the cancellability that is missing from normal Promises, `svelte-concurrency` also provides a public API that allows you to observe the running state of your task without having to set a single flag manually.
+
+Tasks that live on components are automatically cancelled when their context is destroyed, meaning you don't need to worry about the clean up - we've got you covered.
+
+Choose whether you want to keep the oldest, keep the newest or keep all instances of your task to help boost the performance of your app and reduce unnecessary server load.
+
+## How to use `svelte-concurrency`
+
+Install it using your favourite package manager:
+
+```bash
+	pnpm install svelte-concurrency
+```
+
+Then put it to work immediately wherever you want cancellable promises.
+
+## Task structure
+
+All tasks will return a store with the same structure:
+
+- error: Error - if an error occurred, it will be returned here,
+- results: Array - all of the results from previous invocations of this task,
+- last_successful: Any - the return value from the last successful run of the task
+- is_loading: Boolean - whether the task is currently running on not
```

Comment:
I wonder if we should change the public api to be camelCase but for the meantime is better if the docs reflects the code. Also we should discuss what other derived state we want to add.